collator/src/lib.rs

```diff
@@ -273,12 +273,15 @@ impl<P: 'static, SP: 'static> RelayChainContext for ApiContext<P, SP> where
 	}
 }
 
-/// Run the collator node using the given `service`.
-fn run_collator_node<S, P, Extrinsic>(
+/// Build a collator service with the given `RelayChainContext` and `ParachainContext`
+/// build by the given `BuildParachainContext` and arguments to the underlying polkadot node.
+///
+/// This function does not block. It returns an `AbstractService` that can be used as a future
+pub fn build_collator_service<S, P, Extrinsic, B>(
 	service: S,
 	para_id: ParaId,
 	key: Arc<CollatorPair>,
-	build_parachain_context: P,
+	build_parachain: B,
```

Comment:
maybe more readable to just do `build_parachain: impl FnOnce() -> P`

Reply:
I forgot about this syntax. Thanks!